packages/xml-views-validation/src/validators/elements/cardinality-of-aggregation.ts

```diff
@@ -0,0 +1,87 @@
+import { filter, forEach, sortBy, map } from "lodash";
+import { XMLElement } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { getUI5AggregationByXMLElement } from "@ui5-language-assistant/logic-utils";
+import { CardinalityOfAggregationIssue } from "../../../api";
+
+export function validateCardinality(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): CardinalityOfAggregationIssue[] {
+  const aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  if (aggregation === undefined || xmlElement.syntax.openName === undefined) {
+    return [];
+  }
+
+  const allCousinsElements = getAllICousinsElements(xmlElement);
+  if (allCousinsElements.length > 1 && aggregation.cardinality === "0..1") {
+    const firstCousin = getFirstCousin(allCousinsElements);
+    // The error will not be shown on first cousin element
+    const currentAggregationSubElements = filter(
+      xmlElement.subElements,
+      (_) => !isFirstCousin(_, firstCousin)
+    );
+
+    const cardinalityOfAggregationIssues = map(
+      currentAggregationSubElements,
+      (_): CardinalityOfAggregationIssue => ({
+        kind: "CardinalityOfAggregation",
+        message: `The aggregation '${xmlElement.name}' has cardinality of 0..1 and can only contain one element.`,
+        severity: "error",
+        offsetRange: {
+          start: (_.syntax.openName ?? _.position).startOffset,
+          end: (_.syntax.openName ?? _.position).endOffset,
+        },
+      })
+    );
+
+    return cardinalityOfAggregationIssues;
+  }
+
+  return [];
+}
+
+function getFirstCousin(allCousinsElements: XMLElement[]): XMLElement {
+  const sortedCousinsElements = sortBy(
+    allCousinsElements,
+    (_) => _.syntax.openName ?? _.position
+  );
+
+  return sortedCousinsElements[0];
+}
+
+function isFirstCousin(
+  cousinElement: XMLElement,
+  firstCousin: XMLElement
+): boolean {
+  return (
+    (cousinElement.syntax.openName ?? cousinElement.position).startOffset ===
+    (firstCousin.syntax.openName ?? firstCousin.position).startOffset
+  );
+}
+
+/* For example - for the following xml snippet the function will return an array of `Toolbar` elemnt and `tnt:ToolHeader` element
+  `<headerToolbar>
+      <Toolbar></Toolbar>
+  </headerToolbar>
+  <headerToolbar>
+      <tnt:ToolHeader></tnt:ToolHeader>
+  </headerToolbar>`
+*/
+function getAllICousinsElements(xmlElement: XMLElement): XMLElement[] {
```

Comment:
looking at the code I am no longer sure it is `cousins`

- we start with XMLElement X
- we go to X.parent
- We collect X.subElements.[...]
- So these do not seem like cousins of X.

Maybe we need another term to describe this

Reply:
Maybe like `getAllAggregattionElementsAtSameDepth`

and then give more meaningful names to the variables/params, e.g (random thoughts)
xmlElement -> `startAggregation`
subElements -> `subElementsOfContainingClass`

I am sure better names can be found, the point is that if we do
something such as:
- `const parent = xmlElement.parent;`
- `filter( parent.subElements,(subElement) =>`

We do not actually add more information on what these mean in the **current function context**. Normally we don't need to add said information, but when we do none trivial logic it may help.